src/mindmap.js

```diff
@@ -620,57 +655,80 @@ FilesMindMap.Util = {
     },
 };
 
-FilesMindMap.NewFileMenuPlugin = {
+// TODO: move to @nextcloud/files
+function getUniqueName(name, names) {
+	let newName = name
+	let i = 1
+	while (names.includes(newName)) {
+		const ext = extname(name)
+		newName = `${basename(name, ext)} (${i++})${ext}`
+	}
+	return newName
+}
+
+FilesMindMap.registerNewFileMenuPlugin = () => {
+	addNewFileMenuEntry({
```

Comment:
Should we copy how richdocuments is doing it? https://github.com/nextcloud/richdocuments/blob/24ff67aa01eaf5cb2c1be29a7bb11bac9344a099/lib/AppInfo/Application.php#L103-L115